okhttp/src/main/java/io/grpc/okhttp/OkHttpReadableBuffer.java

```diff
@@ -40,9 +40,16 @@ public int readableBytes() {
 
   @Override
   public int readUnsignedByte() {
-    return buffer.readByte() & 0x000000FF;
+    try {
+      fakeEofExceptionMethod(); // Okio 2.x can throw EOFException from readByte()
```

Comment:
Err, a bit ugly...  Why not just put `return buffer.readByte() & 0x000000FF;` into the helper method that does the throw?

Reply:
I also considered `if (false) throw new EOFException()`. I could wrap it, but I'm not wild to add a level of indirection just to make the compiler happy here. While this is ugly, it is clear and it is obvious when it is no longer necessary.

Reply:
Also, with the helper method you really have to read the comment to see that the `throws` was on purpose. Here, it is _very clear_ that something strange is going on while at the same time having no impact on behavior. The comment just explains _why_, not "WATCH OUT!"